Dockerfile

```diff
@@ -0,0 +1,21 @@
+FROM python:3.9-slim
+
+RUN apt-get update && apt-get install -y \
+    libpq-dev gcc && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV PYTHONDONTWRITEBYTECODE 1
+ENV PYTHONUNBUFFERED 1
+
+WORKDIR /app
+
+COPY requirements.txt /app/
+RUN pip install --no-cache-dir -r requirements.txt
+
+COPY . /app/
```

Comment:
Copying the entire context with `COPY . /app/` might include unnecessary files. Ensure your `.dockerignore` is properly configured to exclude files that are not needed in the Docker image. This can help reduce the image size and improve security.